### PR TITLE
Implement deblending properly

### PR DESCRIPTION
**Pull request by esheldon**

### CODE REVIEW

File: metadetect/lsst_measure.py [hunk 1]
```diff
@@ -163,12 +173,13 @@ def _do_measure(
     for i, source in enumerate(sources):
 
         # Skip parent objects where all children are inserted
+        # print([s for s in sources.getChildren(source)])
```
Comment (beckermr) [inline metadetect/lsst_measure.py:176]: maybe remove this?

File: metadetect/lsst_metadetect.py [hunk 2]
```diff
@@ -88,13 +91,21 @@ def run_metadetect(
             assert len(mbobs) == 1, 'no multiband for now'
             assert len(mbobs[0]) == 1, 'no multiepoch'
             obs = mbobs[0][0]
+
             exposure = obs.exposure
+
+            if config['deblend']:
+                noise_image = obs.meta['extra_noise_exp'].image
```
Comment (beckermr) [inline metadetect/lsst_metadetect.py:98]: This is definitely monkey patching to some extent. We could add another property to the ngmix obs.

**Review by beckermr — COMMENTED** [review]
Some comments above.

**Comment by beckermr** [comment]
Is there any reason we shouldn't generate the noise for the replacer from the input obs always? Why do we want to be able to pass it in?

**Comment by esheldon** [comment]
I'm trying to make the code in lsst_measure.py as independent of metacal assumptions, e.g. so Eli can run it.

If you don't send a noise image (or seed) it generates the noise deterministically based off the exposure id.

**Comment by esheldon** [comment]
Also it is being generated from the input, so I'm not totally following your question

**Comment by esheldon** [comment]
See this comment for some explanation:  https://github.com/esheldon/metadetect/pull/124/commits/ad552f72d8dfff67ac626f8d5c94112fcefae2aa

**Comment by esheldon** [comment]
Note this noise field is never sheared, it is added as part of the deblending that occurs on a sheared image

File: metadetect/lsst_metadetect.py (hunk 2 above)
Comment (esheldon) [reply metadetect/lsst_metadetect.py:98]: If you prefer we could create a list of lists of these and use them as needed.

Comment (esheldon) [reply metadetect/lsst_metadetect.py:98]: But that has the issue of keeping those lists in sync.

The draw back of a new obs property is scope creep or feature creep, however you want to call it

Comment (beckermr) [reply metadetect/lsst_metadetect.py:98]: Right. My question is why can't we generate this noise image internally?

Comment (esheldon) [reply metadetect/lsst_metadetect.py:98]: We are using the same noise image for all metacal images (see comments below and link)

Thus we would be regenerating the same noise field five times

Comment (beckermr) [reply metadetect/lsst_metadetect.py:98]: Ack NVM. You are making it internally. ignore this.

**Review by beckermr — APPROVED** [review]

**Comment by esheldon** [comment]
Thanks for the review.  Its often good to be forced to justify a change.

**Pull request merged**